index.js

```diff
@@ -336,12 +336,40 @@ function writeBuffer (data) {
   return this._len < this._hwm
 }
 
-function flush () {
+function callFlushCallbackOnDrain (cb) {
+  const onDrain = () => {
+    cb()
+    this.off('error', onError)
+  }
+  const onError = (err) => {
+    cb(err)
+    this.off('drain', onDrain)
+  }
+  this.once('drain', onDrain)
+  this.once('error', onError)
+}
```

Comment:
This is actually not enough to know if the log was actually written, you'll need to call fsync as well.

Reply:
thanks, but only after drain, right?

Reply:
so added fsync but only if `this._fsync` is false to avoid double fsync